moninag/tests/unittests/checks/test_views.py

```diff
@@ -0,0 +1,392 @@
+            email='sven.johanson@gmail.com',
+            is_active=True,
+        )
+
+        self.user1.set_password('password')
+        self.user1.save()
+
+        Server.objects.create(
+            id=1,
+            name='TestServer',
+            address='address1',
+            state='UA',
+            user=CustomUser.objects.get(id=1)
+        )
+
+        Server.objects.create(
+            id=2,
+            name='TestServer user 2',
+            address='address2',
+            state='US',
+            user=CustomUser.objects.get(id=2)
+        )
+
+        NagPlugin.objects.create(
+            id=10,
+            name='TestPlugin',
+            template='TestTemplate'
+        )
+
+        NagPlugin.objects.create(
+            id=20,
+            name='TestPlugin_2',
+            template='TestTemplate_2'
+        )
+
+        Service.objects.create(
+            id=1,
+            name='TestService',
+            status='ok',
+            server=Server.objects.get(id=1)
+        )
+
+        Service.objects.create(
+            id=2,
+            name='TestService_2',
+            status='fail',
+            server=Server.objects.get(id=2)
+        )
+
+        Check.objects.create(
+            id=11,
+            name='TestCheck',
+            plugin=NagPlugin.objects.get(id=10),
+            status=True,
+            last_run='2017-04-01 23:13',
+            target_port=3000,
+            run_freq=10,
+            service=Service.objects.get(id=1),
+            state=True
+        )
+
+        Check.objects.create(
+            id=12,
+            name='TestCheck 2',
+            plugin=NagPlugin.objects.get(id=10),
+            status=True,
+            last_run='2017-04-01 23:13',
+            target_port=2342,
+            run_freq=23,
+            service=Service.objects.get(id=2),
+            state=True
+        )
+
+        self.client = Client()
+        self.client.login(username='bob.johnson@gmail.com', password='password')
+
+    def test_get_check_not_found(self):
+        """Ensure that GET method returns check not found message and 404 status."""
+
+        expected_json = json.dumps({
+            'error': 'Check with specified id was not found.'
+        })
+
+        url = reverse('check', args=[45])
+        response = self.client.get(url)
+
+        received_json = response.content.decode('utf-8')
+
+        self.assertEqual(response.status_code, 404)
+        self.assertJSONEqual(received_json, expected_json)
+
+    def test_get_all_checks(self):
+        """Ensure that GET method returns list of checks
+        if no check id was passed as parameter to the url."""
+
+        url = reverse('check')
+
+        response = self.client.get(url)
+        received_json = response.content.decode('utf-8')
+
+        expected_response = {}
+
+        checks = Check.objects.filter(service__server__user__id=1)
+        expected_response['response'] = [check.to_dict() for check in checks]
```

Comment:
You should consider adding `to_dict()` method as an helper function since its Unit Tests and you are testing **check view**, so in another words, its slightly wrong to use method from **check model** since its another **Unit**. Get the exact copy of model method and add it as helper function so it will be overriden